nemo/core/classes/exportable.py

```diff
@@ -216,21 +222,61 @@ def _export(
                 elif format == ExportFormat.ONNX:
                     # dynamic axis is a mapping from input/output_name => list of "dynamic" indices
                     if dynamic_axes is None:
-                        dynamic_axes = get_dynamic_axes(self.input_module.input_types_for_export, input_names)
-                        dynamic_axes.update(get_dynamic_axes(self.output_module.output_types_for_export, output_names))
-                    torch.onnx.export(
-                        jitted_model,
-                        input_example,
-                        output,
-                        input_names=input_names,
-                        output_names=output_names,
-                        verbose=verbose,
-                        do_constant_folding=do_constant_folding,
-                        dynamic_axes=dynamic_axes,
-                        opset_version=onnx_opset_version,
-                        keep_initializers_as_inputs=keep_initializers_as_inputs,
-                        export_modules_as_functions=export_modules_as_functions,
-                    )
+                        dynamic_axes = self.dynamic_shapes_for_export(use_dynamo)
```

Comment:
Well the whole default functionality was and still is available in utility function - that's what the default implementation of this method does. I explicitly wanted to add this method so that network can overload it with more meaningful dynamic shapes than the default inference from types can provide. For example, there is a way to express that 'batch' and 'length' dimensions are the same across all 3 inputs etc - that really helps especially with dynamo export.

Reply:
Makes sense. LGTM then